src/socket.rs

```diff
@@ -73,6 +89,23 @@ mod linux {
     use socket2::{SockAddr, SockRef};
     use tokio::io;
 
+    pub fn set_ip_freebind(sock: &SockRef) -> io::Result<()> {
```

Comment:
You should use [`socket2::set_freebind`](https://docs.rs/socket2/latest/socket2/struct.Socket.html#method.set_freebind) instead of reimplementing (see: #235).

Also, you need to use[`socket2::set_freebind_ipv6`](https://docs.rs/socket2/latest/socket2/struct.Socket.html#method.set_freebind_ipv6) for IPv6.

Reply:
Thanks for sharing the info. Updated, but set IPV6_TRANPARENT option is not available currently, so keep its implementation in our mod.

Reply:
I believe that `IP_TRANSPARENT` works for both IPv4 and IPv6, so `IPV6_TRANPARENT` might not be necessary. But it would be good to verify.